src/libbson/src/bson/bson-utf8.c

```diff
@@ -264,77 +376,104 @@ bson_utf8_escape_for_json (const char *utf8, /* IN */
    bson_unichar_t c;
    bson_string_t *str;
    bool length_provided = true;
+   ssize_t pos;
    const char *end;
 
    BSON_ASSERT (utf8);
 
-   str = bson_string_new (NULL);
-
    if (utf8_len < 0) {
       length_provided = false;
       utf8_len = strlen (utf8);
    }
 
+   if (utf8_len == 0) {
+      return bson_strdup ("");
+   }
+
    end = utf8 + utf8_len;
 
-   while (utf8 < end) {
+   str = bson_string_alloc (utf8_len);
```

Comment:
```suggestion
   str = bson_string_alloc (utf8_len);
```
Consider using a local `size_t utf8_ulen` variable after performing a bounds check on the result of the `strlen (utf8)` call above. This will address most of the new `-Wsign-conversion` warnings in this PR.

See other comment regarding `size_t` parameter for `bson_string_alloc`.